[RadioGroup] Add inline prop for radio button group

cathyxz opened the pull request:
#### Fixes #1399

#### Checklist

- [x] [Enable CircleCI for your fork](https://circleci.com/add-projects)
- [x] Update documentation

#### Changes proposed in this pull request:

Add `inline` as a prop to RadioGroup to enable inline radio buttons. 

#### Screenshot

<img width="287" alt="screen shot 2017-08-20 at 9 15 25 pm" src="https://user-images.githubusercontent.com/1528181/29503820-e67e8ee4-85ef-11e7-9aa1-5f400ed066bb.png">

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/palantir/blueprint/1475)
<!-- Reviewable:end -->

giladgray commented on packages/core/src/components/forms/radioGroup.tsx:
```diff
@@ -89,11 +94,12 @@ export class RadioGroup extends AbstractComponent<IRadioGroupProps, {}> {
         return {
             checked: value === this.props.selectedValue,
             disabled: disabled || this.props.disabled,
+            inline: inline == null ? false : inline,
```
no need for a ternary when dealing with booleans. should be safe to just pass `inline` directly, or you could assign default prop value.

giladgray commented on packages/core/src/components/forms/radioGroup.tsx:
```diff
@@ -20,6 +20,11 @@ export interface IRadioGroupProps extends IProps {
     disabled?: boolean;
 
+    /**
+     * True if radio buttons are to be displayed inline horizontally.
```
please match language for other boolean props, like the one above which begins "Whether..." (like you did in the file above)

giladgray commented on packages/core/src/components/forms/controls.tsx:
```diff
@@ -72,6 +75,7 @@ export class Control<P extends IControlProps> extends React.Component<P, {}> {
             typeClassName,
             { [Classes.DISABLED]: this.props.disabled },
             this.props.className,
+            { [Classes.INLINE]: this.props.inline },
```
merge this into the object on line 76, so it has two entries.

cathyxz commented:
@giladgray thanks for the prompt review! Addressed comments. I was going to amend my commit, but I think squash and merge means you can just "squash" it in the Github UI?

giladgray commented:
@cathyxz yes please do not amend commits that have already been pushed--that destroys any semblance of history as you cannot see my comments on commits that no longer exist. we use squash&merge to keep a clean `master`
